lib/client/connection.js

```diff
@@ -714,3 +692,43 @@ Connection.prototype._handleSnapshotFetch = function(error, message) {
   delete this._snapshotRequests[message.id];
   snapshotRequest._handleResponse(error, message);
 };
+
+Connection.prototype._handleLegacyInit = function(message) {
+  // If the minor protocol version has been set, we can ignore this legacy
+  // init message, and continue with the updated handshake protocol.
+  if (message.protocolMinor) return this._initializeHandshake();
```

Comment:
Let's move the `_initializeHandshake()` call to initial connect time, so that it's not dependent on receiving the legacy init message from the server. That way, in the future, we can remove the legacy init message without breaking "protocol 1.1" clients.